internal/catalog/internal/types/validators.go

```diff
@@ -10,9 +10,10 @@ import (
 	"regexp"
 	"strings"
 
-	"github.com/hashicorp/go-multierror"
 	"google.golang.org/protobuf/proto"
 
+	"github.com/hashicorp/go-multierror"
```

Comment:
This is a surprising sort of formatting of the import blocks. I would've figured it would be something like:
```
- stdlib
- 3rdparty
- 1stparty (to the project)
```

Reply:
I'm glad someone noticed! Yes, this is a nuanced difference but I'm fine with whatever is less surprising.  To be clear, currently the order is:

Current:
```
- stdlib
- 3rd party
- hashicorp 1st party
- consul only
```

RB's Proposal:
```
- stdlib
- 3rd party
- hashicorp 1st party & consul
```

Thoughts?

Reply:
I missed noticing layer3 was hashicorp. I was extrapolating from a myopic part of one diff sadly.

My local dev has always been configured as:
- go stdlib
- everything not in the current module/project (3rdparty and hashicorp-non-consul)
- this project (consul)

As that was easy to do with `goimports -local github.com/hashicorp/consul` originally. I was not proposing mixing the final tier to be a combo of consul+general hashicorp.

I guess I'm cool with either of:
**current**
- go stdlib
- 3rdparty
- hashicorp-not-consul
- consul

**my historical approach**
- go stdlib
- 3rdparty+hashicorp-non-consul (2+3 above)
- consul

Do the submodules get treated as part of "consul" or separately?

Reply:
I went ahead and updated gci config to 3 sections as proposed. Diff in https://github.com/hashicorp/consul/pull/19634/commits/33912ed62fdcbea92d2612a87220cb6b8bce682e

Reply:
I guess in that case I have no issues with the 4 block version you initially spiked.

Reply:
Well, I spoke too soon, `3rdparty+hashicorp-non-consul (2+3 above)` is possible. Diff in https://github.com/hashicorp/consul/pull/19634/commits/6a193e39ae3f132dec2eb5b64493be25227492f7